src/desktop/lib/psi/buildServerApp.js

```diff
@@ -0,0 +1,87 @@
+import React, { Fragment } from 'react'
+import ReactDOMServer from 'react-dom/server'
+import createRender from 'found/lib/createRender'
+import queryMiddleware from 'farce/lib/queryMiddleware'
+import serialize from 'serialize-javascript'
+import { AppProvider } from './AppProvider'
+import { Resolver } from 'found-relay'
+import { getFarceResult } from 'found/lib/server'
+import { getLoadableState } from 'loadable-components/server'
+
+import { createRelayEnvironment } from './relayEnvironment'
+
+export function buildServerApp(routeConfig, url = '/') {
+  return new Promise(async (resolve, reject) => {
+    try {
+      if (!process.env.SSR_ENABLED) {
+        return resolve({
+          RelayApp: x => x,
+          relayData: {},
+          loadableScripts: {},
+        })
+      }
+
+      const environment = createRelayEnvironment()
+      const historyMiddlewares = [queryMiddleware]
+      const resolver = new Resolver(environment)
+      const render = createRender({})
+
+      const { redirect, status, element } = await getFarceResult({
+        url,
+        historyMiddlewares,
+        routeConfig,
+        resolver,
+        render,
+      })
+
+      // TODO: Deal with redirect one level up?
+      if (redirect) {
+        return resolve({ redirect, status })
+      }
+
+      const App = () => (
+        <AppProvider
+          provide={{
+            environment,
+            resolver,
+            routeConfig,
+          }}
+        >
+          {element}
+        </AppProvider>
+      )
+
+      // Kick off relay requests
+      ReactDOMServer.renderToString(App)
+
+      const relayData = await environment.relaySSRMiddleware.getCache()
+      const loadableState = await getLoadableState(App())
+
+      resolve({
+        App: ({ children, ...props }) => (
+          <div>
+            <div
+              dangerouslySetInnerHTML={{
+                __html: `
+                  ${loadableState.getScriptTag()}
+
+                  <script>
+                    var __RELAY_BOOTSTRAP__ = ${serialize(
+                      JSON.stringify(relayData),
+                      { isJSON: true }
+                    )};
+                  </script>
```

Comment:
I like this approach to "self mounting" ssr (meaning, the dev implementing an SSR route doesn't have to worry about bootstrapping data or putting it somewhere to go down the wire) but it leads to an HTML mismatch warning on rehydration client-side -- gonna look into fixing that